linodecli/cli.py

```diff
@@ -205,3 +220,89 @@ def user_agent(self) -> str:
             f"linode-api-docs/{self.spec_version} "
             f"python/{version_info[0]}.{version_info[1]}.{version_info[2]}"
         )
+
+    @staticmethod
+    def _load_openapi_spec(spec_location: str) -> OpenAPI:
+        """
+        Attempts to load the raw OpenAPI spec (YAML or JSON) at the given location.
+
+        :param spec_location: The location of the OpenAPI spec.
+                              This can be a local path or a URL.
+
+        :returns: A tuple containing the loaded OpenAPI object and the parsed spec in
+                  dict format.
+        """
+
+        with CLI._get_spec_file_reader(spec_location) as f:
+            parsed = CLI._parse_spec_file(f)
+
+        return OpenAPI(parsed)
+
+    @staticmethod
+    @contextlib.contextmanager
+    def _get_spec_file_reader(
+        spec_location: str,
+    ) -> ContextManager[IO]:
```

Comment:
Given the new OpenAPI spec is so large, I used a ContextManager here to avoid loading the spec into memory more times than we need to.